### PR TITLE
analyze: merge task and send task concurrently

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref #47275

Problem Summary:

### What is changed and how it works?

Start the merge operation early to reduce memory consumption.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: executor/analyze_col_v2.go [hunk 1]
```diff
@@ -270,42 +271,52 @@ func (e *AnalyzeColumnsExecV2) buildSamplingStats(
 	// Start workers to merge the result from collectors.
 	mergeResultCh := make(chan *samplingMergeResult, statsConcurrency)
 	mergeTaskCh := make(chan []byte, statsConcurrency)
+	taskEg, _ := errgroup.WithContext(context.Background())
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply executor/analyze_col_v2.go]: Great, I have fixed.

File: executor/analyze_col_v2.go [hunk 2]
```diff
@@ -270,42 +272,54 @@ func (e *AnalyzeColumnsExecV2) buildSamplingStats(
...
-		if !ok {
-			break
-		}
-		if mergeResult.err != nil {
-			err = mergeResult.err
-			if isAnalyzeWorkerPanic(mergeResult.err) {
-				mergeWorkerPanicCnt++
+	mergeEg, mergeCtx := errgroup.WithContext(context.Background())
+	mergeEg.Go(func() (err error) {
+		for mergeWorkerPanicCnt < statsConcurrency {
```
Comment (qw4990) [inline executor/analyze_col_v2.go:289]: Should we recover it? Otherwise, if it panics, the whole process will exit.

**Review by qw4990 — APPROVED** [review]

File: executor/analyze_col_v2.go (hunk 2 above)
Comment (hawkingrei) [reply executor/analyze_col_v2.go:299]: Done

**Pull request merged**